tensorboard/data/server/server.rs

```diff
@@ -350,6 +369,44 @@ mod tests {
         );
     }
 
+    #[tokio::test]
+    async fn test_list_plugins_multiple_timeseries_same_type() {
+        let commit = CommitBuilder::new()
+            .scalars("train", "xent2", |b| b.build())
+            .scalars("train", "xent", |b| b.build())
+            .build();
+        let handler = sample_handler(commit);
+        let req = Request::new(data::ListPluginsRequest {
+            experiment_id: "123".to_string(),
+        });
+        let res = handler.list_plugins(req).await.unwrap().into_inner();
+        assert_eq!(
+            res.plugins.into_iter().map(|p| p.name).collect::<Vec<_>>(),
+            vec!["scalars"]
+        );
+    }
+
+    // TODO(@bileschi): Add a test for the case of multiple different types of
+    // tags once sample CommitBuilders exist for plugins beyond "scalar".
+
+    #[tokio::test]
+    async fn test_list_plugins_no_data() {
+        let commit = CommitBuilder::new().build();
+        let handler = sample_handler(commit);
+        let req = Request::new(data::ListPluginsRequest {
+            experiment_id: "123".to_string(),
+        });
+        let res = handler.list_plugins(req).await.unwrap().into_inner();
+        let expected: Vec<String> = vec![];
+        assert_eq!(
+            res.plugins
+                .into_iter()
+                .map(|p| p.name)
+                .collect::<Vec<String>>(),
+            expected
```

Comment:
Can inline `expected` as `Vec::<String>::new()` (and use `collect::<Vec<_>>()` again).

Reply:
Not done?

Reply:
done